File: AndroidTai/app/src/main/java/com/tai/androidtai/modules/mealDetails/MealDetailsPresenter.java
package com.tai.androidtai.modules.mealDetails;

import com.tai.androidtai.modules.core.BaseContract;
import com.tai.androidtai.modules.core.BasePresenter;

public class MealDetailsPresenter extends BasePresenter implements MealDetailsContract.Presenter{

    private final MealDetailsContract.Router mRouter;

    public MealDetailsPresenter(MealDetailsContract.Router router) {
        mRouter = router;
    }

    @Override
    public void subscribe(BaseContract.View view) {
        // Nothing to do
    }

    @Override
    public void unsubscribe(BaseContract.View view) {
        // Nothing to do
    }

    @Override
    public void openMovie(String movieLink) {
        mRouter.openMovie(movieLink);
    }
}

File: AndroidTai/app/src/main/java/com/tai/androidtai/modules/mealDetails/MealDetailsPresenter.kt
package com.tai.androidtai.modules.mealDetails

import com.tai.androidtai.modules.core.BaseContract
import com.tai.androidtai.modules.core.BasePresenter

class MealDetailsPresenter(private val mRouter: MealDetailsContract.Router) : BasePresenter(), MealDetailsContract.Presenter {

    override fun subscribe(view: BaseContract.View) {
        // Nothing to do
    }

    override fun unsubscribe(view: BaseContract.View) {
        // Nothing to do
    }

}
